### ISSUE TITLE
Values types should not be combined with `NIL`

### ISSUE THREAD

**Issue by digikar99** [alex-gutev/cl-form-types#6]
Essentially:

```lisp
CL-USER> (cl-form-types:form-type '(the string (values "" 1.0)) nil)
(VALUES (AND STRING (SIMPLE-ARRAY CHARACTER (0))) (AND NIL (EQL 1.0)))
```

The second values-type should be just `(EQL 1.0)` or `(AND T (EQL 1.0)` rather than `(AND NIL (EQL 1.0))`.

**Comment by alex-gutev** [comment]
I see, I missed this part of the hyperspec: "It is permissible for form to yield a different number of values than are specified by value-type".

**Comment by alex-gutev** [comment]
Fixed in d0c57a6.
